include/rcutils/types/string_map.h

```diff
@@ -273,6 +273,47 @@ RCUTILS_WARN_UNUSED
 rcutils_ret_t
 rcutils_string_map_unset(rcutils_string_map_t * string_map, const char * key);
 
+/// Get whether or not a key exists.
+/**
+ * The key needs to be a null terminated c string.
+ *
+ * This function can fail and return false if the key is not found,
+ * or the string_map is NULL or invalid, or if the key is NULL.
+ * In all cases no error message is set.
+ *
+ * \param[in] string_map rcutils_string_map_t to be searched
+ * \param[in] key map key, must be null terminated c string
+ * \return `true` if key is in the map, or
+ * \return `false` if key is not in the map, or
+ * \return `false` for invalid arguments, or
+ * \return `false` if the string map is invalid, or
+ * \return `false` if an unknown error occurs
```

Comment:
if there are many different cases for this to return false, should we provide a way to know the reason of the failure ? (same below)

Reply:
The unknown error case doesn't actually apply here, so really they're all just from users passing invalid arguments, and I suppose (base off of https://github.com/ros2/rcutils/blob/79b93c20462b0e5ec766cd1807b0643bf25b9824/include/rcutils/types/string_map.h#L314) we prefer the cleaner interface when there aren't many internal failure modes. I removed the "unknown error" return case in 27bde82868c060878390e835d0754f8504d33d13 and think that should be sufficient (until the function grows in complexity underneath)